slurm/slurm.py

```diff
@@ -0,0 +1,108 @@
+import os
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+os.environ["TEST_ENVAR_OUTSIDE"] = 'defined'
+
+affinity_script = """
+import os
+affinity = os.sched_getaffinity(0)
+print("num_cores: " + str(len(affinity)))
+for aff in affinity:
+    print("affinity: " + str(aff))"""
```

Comment:
can't we better let this spew out json?

Reply:
using json is certainly possible but more work to implement.
the thing is that you can have 1 or more tasks running this code, so each task has to write its own json file, and the check has to read each of them separately.
on the other hand, each task writing its own file is cleaner than lumping all output into the log file.